test/test_context.py

```diff
@@ -192,6 +196,37 @@ def test_cartocontext_table_exists(self):
         with self.assertRaises(NameError):
             cc._table_exists(self.test_read_table)
 
+    def test_cartocontext_delete(self):
+        """CartoContext.delete"""
+        cc = cartoframes.CartoContext(base_url=self.baseurl,
+                                      api_key=self.apikey)
+        data = {'col1': [1,2,3],
+                'col2': ['a','b','c']}
+        schema = {'col1': int,
+                  'col2': 'object'}
+        df = pd.DataFrame(data).astype(schema)
+
+        cc.write(df, self.test_delete_table)
+        cc.delete(self.test_delete_table)
+
+        # check that querying recently deleted table raises an exception
+        with self.assertRaises(CartoException):
+            cc.sql_client.send('select * from {}'.format(self.test_delete_table))
+
+        # try to delete a table that does not exists
+        def fxn():
+            cc.delete('non_existent_table')
+
+        with warnings.catch_warnings(record=True) as w:
+            # Cause all warnings to always be triggered.
+            warnings.simplefilter("always")
+            # Trigger a warning.
+            fxn()
```

Comment:
Why not replace `fxn()` with `cc.delete...`?

Reply:
I just copied the format from the warnings testings page. The function that triggers the warning was defined above, so that's how I wrote it!

Reply:
Could you replace `fxn()` with `cc.delete..` to simplify things?